codalab/client/rest_client.py

```diff
@@ -0,0 +1,94 @@
+from contextlib import closing
+import httplib
+import json
+import urllib
+import urllib2
+import urlparse
+
+
+class RestClientException(Exception):
+    """
+    Exception raised by the RestClient methods on error. If
+    client_error is False, the failure is caused by a server-side error and
+    can be retried.
+    """
+
+    def __init__(self, message, client_error):
+        super(RestClientException, self).__init__(message)
+        self.client_error = client_error
+
+
+class RestClient(object):
```

Comment:
As in, keep my separate JsonApiClient implementation for now in this PR? (I was hoping to use the upload_with_chunked_encoding method, so I should just make a copy of it for now?)